haystack/document_stores/faiss.py

```diff
@@ -258,7 +258,7 @@ def write_documents(
             documents=document_objects, index=index, duplicate_documents=duplicate_documents
         )
         if len(document_objects) > 0:
-            add_vectors = False if document_objects[0].embedding is None else True
+            add_vectors = all(doc.embedding is not None for doc in document_objects)
```

Comment:
This will make FAISS behave like Weaviate - we either index every document or none